paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreRead.java

```diff
@@ -104,7 +104,7 @@ public RecordReader<InternalRow> createReader(DataSplit split) throws IOExceptio
         DataFilePathFactory dataFilePathFactory =
                 pathFactory.createDataFilePathFactory(split.partition(), split.bucket());
         List<ConcatRecordReader.ReaderSupplier<InternalRow>> suppliers = new ArrayList<>();
-        if (split.beforeFiles().size() > 0) {
```

Comment:
Here don't need to use isEmpty, size > 0 is also a good way.

Reply:
reverted